commcare_connect/static/js/gtx.js

```diff
@@ -0,0 +1,53 @@
+(() => {
+  // === CONFIG ===
+  const ALLOWED_TAG_TYPES = ['google'];
+  const BLOCKED_TAG_TYPES = ['jsm', 'html', 'img', 'j', 'k'];
+
+  // === INIT DATA LAYER ===
+  window.dataLayer = window.dataLayer || [];
+  window.dataLayer.push({
+    'gtm.start': new Date().getTime(),
+    event: 'gtm.js',
+  });
+
+  const setAllowedTagTypes = () => {
+    window.dataLayer.push({
+      'gtm.allowlist': ALLOWED_TAG_TYPES,
+      'gtm.blocklist': BLOCKED_TAG_TYPES,
+    });
+  };
+
+  const gtmSendEvent = (eventName, eventData = {}, callbackFn) => {
+    const data = { event: eventName, ...eventData };
+    window.dataLayer.push(data);
+    if (typeof callbackFn === 'function') callbackFn();
+  };
+
+  const initializeGTM = () => {
+    const gtmDataScript = document.getElementById('gtm-data');
+    if (gtmDataScript) {
+      const gtmData = JSON.parse(gtmDataScript.textContent);
+      if (gtmData.gtmID) {
+        window.dataLayer.push({
+          userEmail: gtmData.userEmail,
```

Comment:
@hridaykedia Can you please remove this userEmail data?
We are trying not send any personal identifiers for users to Google Analytics.